src/main/java/org/entando/entando/aps/system/services/security/NonceInjector.java

```diff
@@ -0,0 +1,50 @@
+package org.entando.entando.aps.system.services.security;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class NonceInjector {
+    private static final String NONCE_INJECTION = "nonce=\"<@wp.cspNonce />\"";
+
+    public static String process(String source) {
+        if (source == null) return null;
+
+        Pattern pattern = Pattern.compile("(<script)(?:[\\t\\r\\n\\s]+([^\"=]*)=\"([^\"]*)\"[\\t\\r\\n\\s]*)*([^>]*>)");
```

Comment:
@ffleandro Pattern.compile is thread save and produces an immutable result.
You may put this in a final static constant.